ConTabs/OutputBuilder.cs

```diff
@@ -140,6 +156,44 @@ private void DataLine(List<CellParts> parts, int line)
                 }
             }
 
+            private int GetWidthOfPaddingAndBorders()
+            {
+                return table._colsShown.Count * table.Padding.GetHorizontalPadding() + table._colsShown.Count + 1;
+            }
+
+            private int GetTableWidthAfterApplyingStretchStyles()
+            {
+                int colWidths = 0;
+                foreach (Column column in table._colsShown)
+                {
+                    int colWidth = table.TableStretchStyles.CalculateOptimalWidth(column);
+                    column.LongStringBehaviour.DisplayWidth = colWidth;
+                    colWidths += colWidth;
+                }
+                int realWidth = colWidths + GetWidthOfPaddingAndBorders();
+                int adaptedWidth = table.TableStretchStyles.CalculateAdditionalWidth(table._colsShown, realWidth, table.CanvasWidth);
+                return adaptedWidth + GetWidthOfPaddingAndBorders();
+            }
+
+            private int GetOptimalTableWidth(List<Column> hiddenCols)
+            {
+                int tableWidth = GetTableWidthAfterApplyingStretchStyles();
+
+                while (table.CanvasWidth < tableWidth && table._colsShown.Count > 0)
+                {
+                    var columnToHide = table._colsShown[table._colsShown.Count - 1];
+                    hiddenCols.Add(columnToHide);
+                    tableWidth -= columnToHide.LongStringBehaviour.DisplayWidth + table.Padding.GetHorizontalPadding() + 1;
+                    columnToHide.Hide = true;
```

Comment:
As you allude to in your PR, re-using the `Hide` property doesn't quite feel right. Hard to put my finger on it, but it feels like we're doing something different here. Perhaps we could add an internal property called "dropped" or "suppressed", or something?

Reply:
Agreed. Will do.